Source/Core/VideoCommon/PixelEngine.cpp

```diff
@@ -273,26 +251,22 @@ static void SetFinish_OnMainThread(u64 userdata, s64 cyclesLate)
 // THIS IS EXECUTED FROM VIDEO THREAD
 void SetToken(const u16 _token, const int _bSetTokenAcknowledge)
 {
+  INFO_LOG(PIXELENGINE, "VIDEO Backend raises INT_CAUSE_PE_TOKEN (btw, token: %04x)", _token);
+  s_Token.store(_token);
```

Comment:
I'm not sure about using an atomic to transmit the token to the CPU thread, it creates a race condition where the token can arrive before the interrupt. It might even be possible for a new token to arrive before the cpu thread even gets around to acknowledging the interrupt. 

It might be a better idea to communicate the token via the event's argument, so both the token and the interrupt (if applicable) arrive at the same time.